### ISSUE TITLE
The CFG is not a DAG, as it may contain loops

### ISSUE THREAD

**Issue by jkimblad** [jkimblad/speeddemon#2]
example:

```
speeddemon s = speeddemon();
s.init()
for (int i = 0; i < 10; i++) {
    s.time_stamp(1);
    foo();
    bar();
    s.time_stamp(2);
}
```

**Comment by jkimblad** [comment]
We need to make sure loops are handled correctly by creating a loop in the graph and not a continous chain of alternating ID's

**Comment by jkimblad** [comment]
Fixed by b930dc0316257c5b0f1e957ced4e6b8b2b7b310a
